### ISSUE TITLE
AppCenter Retirement: How to handle .NET MAUI Apps?

### ISSUE THREAD

**Issue by thomasgalliker** [microsoft/appcenter#2625]
Visual Studio App Center is scheduled for retirement on March 31, 2025. [Learn more about support timelines and recommended alternatives.](https://aka.ms/appcenter/retire)

What is your proposed migration path for Xamarin Forms and .NET MAUI apps that use the AppCenter features Build, Analytics, Crashes, Distribution? Please post your ideas below.

**Comment by huyphams** [comment]
Please open source self-hosting appcenter.
https://github.com/microsoft/appcenter/issues/2623

**Comment by Xynratron** [comment]
We integrated the **Builds** completly in our **Azure Devops Pipelines**.

**Application Insights** seems not to be a good replacement for **Analytics / Crashes** - any other Ideas?

The **Distribution** is the biggest Problem, because Microsoft Partner Center Apps & Games is a mess and not updated in years (as far as I can see). After anouncing the retirement for Microsoft AppStore for Business, what ist the roadmap for distributing Windows Applications?

**Comment by thomasgalliker** [comment]
Does anyone have experience with aptabase? It seems to have tracking features. Not sure about diagnostics/crash analysis. It's open-source and free for self-hosting instances.
[https://github.com/aptabase](https://github.com/aptabase)

**Comment by LathanHarper** [comment]
It's the diagnostic logs I will miss the most!

**Comment by LathanHarper** [comment]
> We integrated the **Builds** completly in our **Azure Devops Pipelines**.
> 
> **Application Insights** seems not to be a good replacement for **Analytics / Crashes** - any other Ideas?
> 
> The **Distribution** is the biggest Problem, because Microsoft Partner Center Apps & Games is a mess and not updated in years (as far as I can see). After anouncing the retirement for Microsoft AppStore for Business, what ist the roadmap for distributing Windows Applications?

for Enterprise distribution the InTune product is actually really awesome at that.  When a device (or PC) is enrolled you get your 
own "store" plus all the normal remote admin ability.

**Comment by thomasgalliker** [comment]
Does anyone use [Sentry](https://sentry.io/welcome/) for .NET MAUI analytics and/or crash reporting?

**Comment by kristopherwinstanleyRioTinto** [comment]
We are in the same boat.. looking for good options. Considering Dynatrace, Sentry.. some form or using App Insights SDK (doesn't seem like a good option, especially offline, crashs and bad network connectivity)

**Comment by softlion** [comment]
> Does anyone use [Sentry](https://sentry.io/welcome/) for .NET MAUI analytics and/or crash reporting?

Yes me. Work perfectly with Apple App Store apps. Better than appcenter crash. Full C# stack trace.

Also, for building I tried Azure Pipeline, but in the end switched to codemagic.io  
Codemagic supports the latest macs (M1 M2) and the latest xcode versions. They release new versions fast, you can still select older versions.  
M1 are free. Build time is 10mn total.

**Comment by thomasgalliker** [comment]
Good news. I migrated my MAUI app from AppCenter to Sentry. It‘s pretty cool! We mainly use the crash reporting and user tracking part.
All distribution related task will be done with native Android/iOS tools (question: does anyone have scripts to upload apps to the stores?).

Here you have a small MAUI demo app with Sentry integration: https://github.com/hsrcasnet/MauiSamples/tree/main/MonitoringDemo/MonitoringDemo

**Comment by AlleSchonWeg** [comment]
Hi,
has somebody tested Application Insights with Android and iOS?
My questions are:
- Are crashes handled the same like App Center?
- Are C# StackTraces included in Android and iOS?
- Is Xamarin Forms also supported?

Thank you.

**Comment by r0bbie** [comment]
For building - Azure DevOps Pipelines or GitHub Actions with [Fastlane](https://fastlane.tools/) (latter automating key steps like signing apps, as well as automated app store deployment) are solid choices.

For crashes - a number of choices out there but I'd suggest [Sentry](https://blog.sentry.io/visual-studio-app-center-retirement-why-sentry-is-your-next-step/).

For beta distribution - [Buildstash](https://buildstash.com/) was designed to be a solid replacement for App Center's distribution groups capabilities, and has integrations for Azure Pipelines, GitHub Actions, and Fastlane (disclaimer - I'm working on that tool!)
